### PR TITLE
Merging changes from Paul's fork - Changes to Future Implementations 

### PR DESCRIPTION
**Pull request by lcarrazana**
- Changes to K8s future implementations

Relates to #265

### CODE REVIEW

File: service_packs/kubernetes/container_registry_access/container_registry_access.feature [hunk 1]
```diff
@@ -1,27 +1,23 @@
 @k-cra
 @probes/kubernetes/container_registry_access
 Feature: Protect image container registries
-    As a Security Auditor
-    I want to ensure that containers image registries are secured in my organisation's Kubernetes clusters
-    So that only approved software can be run in our cluster in order to prevent malicious attacks on my organization
+  As a Security Auditor
+  I want to ensure that containers image registries are secured in my organisation's Kubernetes clusters
```
Comment (iantivey) [inline service_packs/kubernetes/container_registry_access/container_registry_access.feature:5]: Can we change this to `I want to ensure that access to container image registries from the Kubernetes cluster is controlled`

File: service_packs/kubernetes/container_registry_access/container_registry_access.feature [hunk 2]
```diff
@@ -1,27 +1,23 @@
...
+    Background:
+      Given a Kubernetes cluster is deployed
+
     @k-cra-001
-    Scenario: Ensure container image registries are read-only
-        Given a Kubernetes cluster is deployed
-        And I am authorised to pull from a container registry
-        When I attempt to push to the container registry using the cluster identity
-        Then the push request is rejected due to authorization
+    Scenario: Ensure the cluster service account has read only access to the authorized container registry
```
Comment (iantivey) [inline service_packs/kubernetes/container_registry_access/container_registry_access.feature:14]: I think we should move this scenario to "future implementations".  If we implement it we'll need to think quite carefully about how we supply and manage credentials.

File: service_packs/kubernetes/container_registry_access/container_registry_access.feature [hunk 3]
```diff
@@ -1,27 +1,23 @@
...
-
-    @k-cra-003
     Scenario: Ensure deployment from an unauthorised container registry is denied
-        Given a Kubernetes cluster is deployed
-        When a user attempts to deploy a container from an unauthorised registry
-        Then the deployment attempt is denied
+      Given a user attempts to deploy a container from an authorised registry
+      And the deployment attempt is allowed
+      When a user attempts to deploy a container from an unauthorised registry
+      Then the deployment attempt is denied
```
Comment (iantivey) [inline service_packs/kubernetes/container_registry_access/container_registry_access.feature:23]: I like how this one reads. Perhaps we should use it as a reference for some of our other tests.

File: service_packs/kubernetes/container_registry_access/container_registry_access.feature (hunk 1 above)
Comment (lcarrazana) [reply service_packs/kubernetes/container_registry_access/container_registry_access.feature]: This change request for Container Registry Access probe was documented here: https://github.com/citihub/probr/issues/327

File: service_packs/kubernetes/container_registry_access/container_registry_access.feature (hunk 2 above)
Comment (lcarrazana) [reply service_packs/kubernetes/container_registry_access/container_registry_access.feature]: This change request was documented in this issue: #328

**Review by eddie-knight — APPROVED** [review]

**Pull request merged**